xtext-website/documentation/340_lsp_support.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: documentation
+title: LSP Support
+part: Reference Documentation
+---
+
+# {{page.title}} {#lsp-support}
+
+Xtext supports the generation of language servers that comply to the [Language Server Protocol (LSP)](https://microsoft.github.io/language-server-protocol/).
+
+## Getting Started {#getting-started}
+
+**Step 1:** Create a new Xtext Project with language server support.
+![](images/LSP_1_Xtext_Wizard.jpg)
+
+**Step 2:** Implement the Domainmodel language following the [Xtext - 15 Minutes Tutorial](https://www.eclipse.org/Xtext/documentation/102_domainmodelwalkthrough.html).
+
+**Step 3:** Try out the Domainmodel language server in Eclipse based on [LSP4E](https://projects.eclipse.org/projects/technology.lsp4e).
+
+ 1. Configure a content type to the `*.dmodel` files:
+![](images/LSP_2_DomainmodelContentType.jpg)
+
+ 1. Create a launch configuration that starts the Domainmodel language server:
+![](images/LSP_3_DomainmodelLanguageServerLauncher.jpg)
+
+ 1. Assign the Domainmodel content type to the Domainmodel language server launcher:
+![](images/LSP_4_DomainmodelLanguageServer.jpg)
+
+ 1. LSP does not support syntax highlighting (usually it is done on the client side). The following [TextMate](https://projects.eclipse.org/proposals/tm4e-textmate-support-eclipse-ide) json file adds syntax highlighting support for the keywords, single line comments and multi-line comments of the Domainmodel language:
+ ```json
+{
+	"name": "Domainmodel",
+	"scopeName": "text.dmodel",
+	"fileTypes": [
+		"dmodel"
+	],
+	"repository": {
+		"general": {
+			"patterns": [
+				{
+					"include": "#linecomment"
+				},
+				{
+					"include": "#blockcomment"
+				},
+				{
+					"include": "#keyword"
+				}
+			]
+		},
+		"linecomment": {
+			"name": "comment.line.double-dash.dmodel",
+			"begin": "(^[ \\t]+)?(?=//)",
+			"end": "(?=$)"
+		},
+		"blockcomment": {
+			"name": "comment.block.dmodel",
+			"begin": "/\\*(\\*)?(?!/)",
+			"end": "\\*/"
+		},
+		"keyword": {
+			"name": "keyword.control.mydsl",
+			"match": "\\b(entity|datatype)\\b|!"
+
+		}
+	},
+	"patterns": [
+		{
+			"include": "#general"
+		}
+	],
+	"uuid": "8383e49a-fa0d-4bb5-827b-10e8abb294ca"
+}
+```
+ 1. Open an `*.dmodel` file with the Generic Editor and explore the supported language features like syntax highlighting, content assist, validation, quickfixes, formatting ... :
+![](images/LSP_5_Domainmodel_Content_Assist.jpg)
+
+## Language Features {#language-features}
+
+Currently, Xtext supports the following LSP language features:
+
+<table class="table table-bordered">
```

Comment:
this is an unusal markdown table

Reply:
I created that table analog to the [Xtext Web Support table](https://raw.githubusercontent.com/eclipse/xtext/website-published/xtext-website/documentation/330_web_support.md)

Reply:
normal markdown tables dont work?

Reply:
maybe cause of the stupid css checkmark thing which is not accessible at all

Reply:
Well default markdown doesn't have tables so it'll depend on what flavor of MD you're using